pkg/framework/coremodel/staticregistry/provide.go

```diff
@@ -3,15 +3,28 @@ package staticregistry
 import (
 	"github.com/grafana/grafana/pkg/coremodel/dashboard"
 	"github.com/grafana/grafana/pkg/framework/coremodel"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
 )
 
 // ProvideRegistry provides a simple static Registry for coremodels.
 // Coremodels have to be manually added.
 // TODO dynamism
 func ProvideRegistry(
+	feat featuremgmt.FeatureManager,
```

Comment:
Glaring issue - by doing the feature toggle check only in the registry we create of all coremodels, it means a `func Provide*` that takes a `*coremodel.Registry` (as we do in this PR) will correctly receive stubbed-out `coremodel.Interface` objects, but things that rely directly on the coremodels instead (e.g. they take a `*dashboard.Coremodel` as input) will _not_ get the stubbed-out object.

Not really sure of a solution here beyond forcing the feature flag into every coremodel implementation, which isn't really a guarantee.